pgmpy/inference/continuous/base.py

```diff
@@ -0,0 +1,396 @@
+import numpy as np
+
+from pgmpy.utils import _check_1d_array_object, _check_length_equal
+
+
+class BaseGradLogPDF(object):
+    """
+    Base class for evaluating gradient log of probability density function/ distribution
+
+    Classes inheriting this base class can be passed as an argument for
+    finding gradient log of probability distribution in inference algorithms
+
+    The class should initialize  self.grad_log and self.log_pdf
+
+    Parameters
+    ----------
+    distribution_param : A 1d array like object (numpy.ndarray or list)
```

Comment:
@khalibartan This is the point in the space at which we want to compute the gradient. Right ? If yes, I think we should have a better name for this.